docs/guides/js-waku/use-waku-react.md

```diff
@@ -0,0 +1,356 @@
+	});
+
+	// "node" is the created Light Node
+	// "error" captures any error that occurs during node creation
+	// "isLoading" indicates whether the node is still being created
+	// Use these to handle states and render the UI in your application
+}
+```
+
+## Create an Encoder and Decoder
+
+Use the `useCreateContentPair()` hook to create a message `encoder` and `decoder` pair:
+
+```js title="App.js"
+import { useCreateContentPair } from "@waku/react";
+
+function App() {
+	// Choose a content topic
+	const contentTopic = "/waku-react-guide/1/message/utf8";
+	const ephemeral = false;
+
+	// Create a message encoder and decoder pair
+	const { encoder, decoder } = useCreateContentPair(contentTopic, ephemeral);
+
+	// "encoder" is the message encoder
+	// "decoder" is the message decoder
+	// Use these to handle the messages in your application
+}
+```
+
+## Send Messages Using Light Push
+
+Use the `useLightPush()` hook to bind `Light Push` methods to a node and `encoder`:
+
+```js title="App.js"
+import {
+	useCreateLightNode,
+	useCreateContentPair,
+	useLightPush,
+} from "@waku/react";
+import { Protocols } from "@waku/interfaces";
+import { utf8ToBytes } from "@waku/sdk";
+
+function App() {
+	// Create and start a Light Node and wait for remote peers
+	const { node, error, isLoading } = useCreateLightNode({
+		options: { defaultBootstrap: true },
+		protocols: [Protocols.LightPush, Protocols.Filter],
+	});
+
+	// Choose a content topic and create an encoder
+	const contentTopic = "/waku-react-guide/1/message/utf8";
+	const { encoder  } = useCreateContentPair(contentTopic);
+
+	// Wait for the node to finish loading before sending messages
+	// (isLoading === false)
+
+	// Bind push method to a node and encoder
+	const { push } = useLightPush({ node, encoder });
+
+	// Send the message using Light Push
+	const sendMessage = async (text) => {
+		if (!push || !text) {
+			return;
+		}
+		const payload = utf8ToBytes(text);
+		await push({ payload });
+	};
+	sendMessage("Hello, World!");
+}
+```
+
+:::info
+Wait for the node to finish loading before sending messages (`isLoading` === `false`).
+:::
+
+## Receive Messages Using Filter
+
+Use the `useFilterMessages()` hook to receive messages from a `Filter` subscription:
+
+```js title="App.js"
+import {
+	useCreateLightNode,
+	useCreateContentPair,
+	useFilterMessages,
+} from "@waku/react";
+import { Protocols } from "@waku/interfaces";
+
+function App() {
+	// Create and start a Light Node and wait for remote peers
+	const { node } = useCreateLightNode({
+		options: { defaultBootstrap: true },
+		protocols: [Protocols.LightPush, Protocols.Filter],
+	});
+
+	// Choose a content topic and create a decoder
+	const contentTopic = "/waku-react-guide/1/message/utf8";
+	const { decoder } = useCreateContentPair(contentTopic);
+
+	// Receive messages from Filter subscription
+	const { error, messages, isLoading } = useFilterMessages({ node, decoder });
```

Comment:
What happen if we receive more messages? I'd expect using a callback here. Cc @weboko

Reply:
Similar to the "button" comment above.

I think we should have a component where we display messages received.

The final result should be something like https://github.com/waku-org/js-waku-examples/tree/master/examples/relay-reactjs-chat but it uses light push + filter + store.

Reply:
> What happen if we receive more messages? I'd expect using a callback here. Cc @weboko

it will be updated reactively though subscription [here](https://github.com/waku-org/waku-ui/blob/dce1bb6301d152b320fa408d8bd07e9cfe21dc66/src/useFilterMessages.ts#L61)